File: 07_gpu_target/problems/01_where_am_i/where_am_i.cpp
#include <cstdio>
#include <omp.h>

int main() {
  printf("on host: omp_is_initial_device() = %d\n", omp_is_initial_device());
  // TODO: 下のブロックの直前に #pragma omp target を1行追加し, ブロックの中身をデバイス(GPU)上で実行させよ. (表示するだけなので map 節は不要)
  {
    printf("inside target: omp_is_initial_device() = %d\n", omp_is_initial_device());
  }
  return 0;
}

File: 07_gpu_target/problems/01_where_am_i/where_am_i.f90
program where_am_i
  use omp_lib
  print "(a,i0)", "on host: omp_is_initial_device() = ", omp_is_initial_device()
  ! TODO: 下の print を !$omp target ... !$omp end target で囲み, デバイス(GPU)上で実行させよ. (表示するだけなので map 節は不要)
  print "(a,i0)", "inside target: omp_is_initial_device() = ", omp_is_initial_device()
  ! TODO: 上で始めた target 領域を閉じる (!$omp end target).
end program where_am_i
